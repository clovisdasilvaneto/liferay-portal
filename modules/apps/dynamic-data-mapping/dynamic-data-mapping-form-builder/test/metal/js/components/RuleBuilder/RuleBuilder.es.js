import '../../components/RuleEditor/__fixtures__/RuleEditorMockField.es';
import dom from 'metal-dom';
import RuleBuilder from 'source/components/RuleBuilder/RuleBuilder.es';

const spritemap = 'icons.svg';
let component;

const baseConfig = {
	dataProviderInstanceParameterSettingsURL: '/o/dynamic-data-mapping-form-builder-data-provider-instances/',
	dataProviderInstancesURL: '/o/data-provider/',
	functionsURL: '/o/dynamic-data-mapping-form-builder-functions/',
	mode: 'view',
	pages: [
		{
			rows: [
				{
					columns: [
						{
							fields: [
								{
									fieldName: 'text1',
									label: 'label text 1'
								},
								{
									fieldName: 'text2',
									label: 'label text 2'
								}
							]
						}
					]
				}
			]
		}
	],
	rules: [
		{
			actions: [
				{
					action: 'require',
					expression: '[x+2]',
					target: 'text1'
				}
			],
			conditions: [
				{
					operands: [
						{
							type: 'field',
							value: 'text1'
						},
						{
							type: 'field',
							value: 'text2'
						}
					],
					operator: 'equals-to'
				}
			],
			['logical-operator']: 'OR'
		}
	],
	spritemap,

	rolesURL: '/o/dynamic-data-mapping-form-builder-roles/',

	url: '/o/dynamic-data-mapping-form-builder-roles/'
};

describe(
	'RuleBuilder',
	() => {
		beforeEach(
			() => {
				dom.enterDocument('<button id="addFieldButton" class="hide"></button>');

				component = new RuleBuilder(baseConfig);

				jest.useFakeTimers();
			}
		);
		afterEach(
			() => {
				const addbutton = document.querySelector('#addFieldButton');

				if (component) {
					component.dispose();
				}

				dom.exitDocument(addbutton);
			}
		);
		it(
			'should render the list of rules when mode is set to view',
			() => {
				component.setState({mode: 'view'});

				jest.runAllTimers();

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should render rule screen editor when click edit button',
			() => {
				const ruleCardEdit = document.querySelector('.rule-card-edit');

				dom.triggerEvent(ruleCardEdit, 'click', {});

				jest.runAllTimers();

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should render rule screen creator when click add button',
			() => {
				const addbutton = document.querySelector('#addFieldButton');

				dom.triggerEvent(addbutton, 'click', {});

				jest.runAllTimers();

				expect(component).toMatchSnapshot();
			}
		);
	}
);