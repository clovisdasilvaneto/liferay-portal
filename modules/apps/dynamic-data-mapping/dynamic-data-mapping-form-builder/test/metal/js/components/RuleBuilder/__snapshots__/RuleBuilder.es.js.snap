// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuleBuilder should render rule screen creator when click add button 1`] = `
<div class="container">
  <div class="form-builder-rule-builder-container">
    <h1 class="form-builder-section-title text-default">rules</h1>
    <div class="liferay-ddm-form-rule-rules-list-container">
      <div class="form-builder-rule-list">
        <ul class="ddm-form-body-content form-builder-rule-builder-rules-list tabular-list-group">
          <li class="list-group-item">
            <div class="clamp-horizontal list-group-item-content">
              <p class="form-builder-rule-builder-rule-description text-default"><b>if </b><span>field </span><span class="label label-lg label-secondary" data-original-title="label text 1" title="label text 1"><span class="text-truncate-inline"><span class="text-truncate">label text 1</span></span></span><b class="text-lowercase"><em> is-equal-to </em></b><span>field </span><span class="label label-lg label-secondary" data-original-title="label text 2" title="label text 2"><span class="text-truncate-inline"><span class="text-truncate">label text 2</span></span></span><br><span><b>require </b></span><span>field </span><span class="label label-lg label-secondary" data-original-title="label text 1" title="label text 1"><span class="text-truncate-inline"><span class="text-truncate">label text 1</span></span></span></p>
            </div>
            <div class="list-group-item-field">
              <div class="card-col-field">
                <div class="dropdown dropdown-action">
                  <div class="ddm-rule-list-settings" data-card-id="0">
                    <div class="dropdown dropdown-action"><button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled" ref="triggerButton" title="actions" type="button"><svg aria-hidden="true" class="lexicon-icon lexicon-icon-ellipsis-v" focusable="false">
                          <title>ellipsis-v</title>
                          <use xlink:href="icons.svg#ellipsis-v"></use>
                        </svg></button></div><button class="component-action dropdown-toggle cursor-pointer" data-rule-index="0"><svg aria-hidden="true" class="lexicon-icon lexicon-icon-ellipsis-v" focusable="false">
                        <title>ellipsis-v</title>
                        <use xlink:href="icons.svg#ellipsis-v"></use>
                      </svg></button>
                  </div>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
<div class="dropdown-menu" ref="menu">
  <ul class="list-unstyled">
    <li><button class="btn dropdown-item btn-sm btn-link" aria-label="edit" type="button">edit</button></li>
    <li><button class="btn dropdown-item btn-sm btn-link" aria-label="delete" type="button">delete</button></li>
  </ul>
</div>
`;

exports[`RuleBuilder should render the list of rules when mode is set to view 1`] = `
<div class="container">
  <div class="form-builder-rule-builder-container">
    <h1 class="form-builder-section-title text-default">rules</h1>
    <div class="liferay-ddm-form-rule-rules-list-container">
      <div class="form-builder-rule-list">
        <ul class="ddm-form-body-content form-builder-rule-builder-rules-list tabular-list-group">
          <li class="list-group-item">
            <div class="clamp-horizontal list-group-item-content">
              <p class="form-builder-rule-builder-rule-description text-default"><b>if </b><span>field </span><span class="label label-lg label-secondary" data-original-title="label text 1" title="label text 1"><span class="text-truncate-inline"><span class="text-truncate">label text 1</span></span></span><b class="text-lowercase"><em> is-equal-to </em></b><span>field </span><span class="label label-lg label-secondary" data-original-title="label text 2" title="label text 2"><span class="text-truncate-inline"><span class="text-truncate">label text 2</span></span></span><br><span><b>require </b></span><span>field </span><span class="label label-lg label-secondary" data-original-title="label text 1" title="label text 1"><span class="text-truncate-inline"><span class="text-truncate">label text 1</span></span></span></p>
            </div>
            <div class="list-group-item-field">
              <div class="card-col-field">
                <div class="dropdown dropdown-action">
                  <div class="ddm-rule-list-settings" data-card-id="0">
                    <div class="dropdown dropdown-action"><button aria-expanded="false" aria-haspopup="true" aria-label="toggle" class="dropdown-toggle btn btn-unstyled" ref="triggerButton" title="actions" type="button"><svg aria-hidden="true" class="lexicon-icon lexicon-icon-ellipsis-v" focusable="false">
                          <title>ellipsis-v</title>
                          <use xlink:href="icons.svg#ellipsis-v"></use>
                        </svg></button></div><button class="component-action dropdown-toggle cursor-pointer" data-rule-index="0"><svg aria-hidden="true" class="lexicon-icon lexicon-icon-ellipsis-v" focusable="false">
                        <title>ellipsis-v</title>
                        <use xlink:href="icons.svg#ellipsis-v"></use>
                      </svg></button>
                  </div>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
<div class="dropdown-menu" ref="menu">
  <ul class="list-unstyled">
    <li><button class="btn dropdown-item btn-sm btn-link" aria-label="edit" type="button">edit</button></li>
    <li><button class="btn dropdown-item btn-sm btn-link" aria-label="delete" type="button">delete</button></li>
  </ul>
</div>
`;
